Example in 'package-sparsepp.R' fails on Mac/Windows

Hello, I've tried testing the example provided in file `package-sparsepp.R` on both Mac and Windows, but it fails as is with this error:
```
file189415136729.cpp:7:22: fatal error: sparsepp.h: No such file or directory
 #include <sparsepp.h>
                      ^
compilation terminated.
```
On both machines I installed `sparsepp` from CRAN. In order to get it working, I had to modify this chunk:
```c++
#include <Rcpp.h>
using namespace std;
using namespace Rcpp;
// drop-in replacement for unordered_map
//#include <unordered_map>
#include <sparsepp.h>
using spp::sparse_hash_map;
```
To be this:
```c++
// [[Rcpp::plugins(cpp11)]]
#include <Rcpp.h>
using namespace std;
using namespace Rcpp;
// drop-in replacement for unordered_map
//#include <unordered_map>
#include <sparsepp/spp.h>
//[[Rcpp::depends(sparsepp)]]
using spp::sparse_hash_map;
```
After making these edits, the example works great. I'm not sure if this is specific to my machines/setups. If you'd like to include these edits in the R file, let me know, I'd be happy to submit a PR.

Looking at the installed package, the directory structure of the header file is `~/sparsepp/include/sparsepp/spp.h`.

Here's my session info from the Windows machine:
```r
devtools::session_info()
```
```
Session info ------------------------------------------------------------------------------------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.4.3 (2017-11-30)
 system   x86_64, mingw32             
 ui       RStudio (1.1.383)           
 language (EN)                        
 collate  English_United States.1252  
 tz       America/Chicago             
 date     2018-03-04                  

Packages ----------------------------------------------------------------------------------------------------------------------------------------------------------------
 package   * version date       source                         
 base      * 3.4.3   2017-12-06 local                          
 compiler    3.4.3   2017-12-06 local                          
 datasets  * 3.4.3   2017-12-06 local                          
 devtools    1.13.5  2018-02-18 CRAN (R 3.4.3)                 
 digest      0.6.15  2018-01-28 CRAN (R 3.4.3)                 
 graphics  * 3.4.3   2017-12-06 local                          
 grDevices * 3.4.3   2017-12-06 local                          
 memoise     1.1.0   2018-01-26 Github (hadley/memoise@611cfad)
 methods   * 3.4.3   2017-12-06 local                          
 Rcpp      * 0.12.15 2018-01-20 CRAN (R 3.4.3)                 
 sparsepp    0.2.0   2017-08-07 CRAN (R 3.4.3)                 
 stats     * 3.4.3   2017-12-06 local                          
 tools       3.4.3   2017-12-06 local                          
 utils     * 3.4.3   2017-12-06 local                          
 withr       2.1.1   2017-12-19 CRAN (R 3.4.3)                 
 yaml        2.1.17  2018-02-27 CRAN (R 3.4.3)
```

Hi. Thanks for reporting. You are right - I overlooked examples during the
last update. Before it was single header file. Will happy to get/review PR.

5 мар. 2018 г. 6:44 пользователь "Chris Muir" <notifications@github.com>
написал:

Hello, I've tried testing the example provided in file package-sparsepp.R
on both Mac and Windows, but it fails as is with this error:

file189415136729.cpp:7:22: fatal error: sparsepp.h: No such file or directory
 #include <sparsepp.h>
                      ^
compilation terminated.

On both machines I installed sparsepp from CRAN. In order to get it
working, I had to modify this chunk:

#include <Rcpp.h>using namespace std;using namespace Rcpp;// drop-in
replacement for unordered_map//#include <unordered_map>
#include <sparsepp.h>using spp::sparse_hash_map;

To be this:

// [[Rcpp::plugins(cpp11)]]
#include <Rcpp.h>using namespace std;using namespace Rcpp;// drop-in
replacement for unordered_map//#include <unordered_map>
#include <sparsepp/spp.h>//[[Rcpp::depends(sparsepp)]]using
spp::sparse_hash_map;

After making these edits, the example works great. I'm not sure if this is
specific to my machines/setups. If you'd like to include these edits in the
R file, let me know, I'd be happy to submit a PR.

Looking at the installed package, the directory structure of the header
file is ~/sparsepp/include/sparsepp/spp.h.

Here's my session info from the Windows machine:

devtools::session_info()

Session info ------------------------------------------------------------------------------------------------------------------------------------------------------------
 setting  value
 version  R version 3.4.3 (2017-11-30)
 system   x86_64, mingw32
 ui       RStudio (1.1.383)
 language (EN)
 collate  English_United States.1252
 tz       America/Chicago
 date     2018-03-04

Packages ----------------------------------------------------------------------------------------------------------------------------------------------------------------
 package   * version date       source
 base      * 3.4.3   2017-12-06 local
 compiler    3.4.3   2017-12-06 local
 datasets  * 3.4.3   2017-12-06 local
 devtools    1.13.5  2018-02-18 CRAN (R 3.4.3)
 digest      0.6.15  2018-01-28 CRAN (R 3.4.3)
 graphics  * 3.4.3   2017-12-06 local
 grDevices * 3.4.3   2017-12-06 local
 memoise     1.1.0   2018-01-26 Github (hadley/memoise@611cfad)
 methods   * 3.4.3   2017-12-06 local
 Rcpp      * 0.12.15 2018-01-20 CRAN (R 3.4.3)
 sparsepp    0.2.0   2017-08-07 CRAN (R 3.4.3)
 stats     * 3.4.3   2017-12-06 local
 tools       3.4.3   2017-12-06 local
 utils     * 3.4.3   2017-12-06 local
 withr       2.1.1   2017-12-19 CRAN (R 3.4.3)
 yaml        2.1.17  2018-02-27 CRAN (R 3.4.3)

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/dselivanov/r-sparsepp/issues/1>, or mute the thread
<https://github.com/notifications/unsubscribe-auth/AE4u3SC_IsdJEEw3dtGaadWtqgIlT95kks5tbKYugaJpZM4Sbpic>
.

--- a/R/package-sparsepp.R
+++ b/R/package-sparsepp.R
@@ -17,12 +17,14 @@
 #' \dontrun{
 #' library(Rcpp)
 #' code = "
+#' // [[Rcpp::plugins(cpp11)]]
 #' #include <Rcpp.h>
 #' using namespace std;
 #' using namespace Rcpp;
 #' // drop-in replacement for unordered_map
 #' //#include <unordered_map>
-#' #include <sparsepp.h>
+#' #include <sparsepp/spp.h>
+#' //[[Rcpp::depends(sparsepp)]]
 #' using spp::sparse_hash_map;
 #' // @export
 #' // [[Rcpp::export]]

--- a/man/sparsepp-package.Rd
+++ b/man/sparsepp-package.Rd
@@ -24,12 +24,14 @@ It aims to achieve the following objectives:
 \dontrun{
 library(Rcpp)
 code = "
+// [[Rcpp::plugins(cpp11)]]
 #include <Rcpp.h>
 using namespace std;
 using namespace Rcpp;
 // drop-in replacement for unordered_map
 //#include <unordered_map>
-#include <sparsepp.h>
+#include <sparsepp/spp.h>
+//[[Rcpp::depends(sparsepp)]]
 using spp::sparse_hash_map;
 // @export
 // [[Rcpp::export]]

